Documentation update: other (#732)
flame-engine/flame: doc/layers.md

# Layers

> Layers are an useful feature that lets you group rendering by context, as well as allow you to

Reviewer: `useful feature` => hopefully all our features are useful haha

Reviewer: `Layers alloy you to group...`

More clear and direct

pre-render things. This enables, for example, rendering parts of your game that don't change much in

> memory, like a background. By doing this, you'll free resources for more dynamic content that needs

Reviewer: `free resources` is a bit unclear here I think as you are basically replacing cpu resources with memory resources (like any cache). I think `free processing time` will cause less confusion as resource can lead people to think memory and that is the opposite of what this is for

to be rendered every loop cycle.

There are two types of layers on Flame: `DynamicLayer` (for things that are moving or changing) and
`PreRenderedLayer` (for things that are static).

## DynamicLayer

> Dynamic layers are layers that are rendered every time that they are draw on the canvas. As the name

Reviewer: ```suggestion
Dynamic layers are recomputed every time that they are drawn on the canvas. As the name
```

Author: Added `drawn`, but not convinced by `recomputed`, I think that is a bit unclear?

Author: Because it is the actual rendering that is grouped, right?

> suggests, it is meant for dynamic content and is most useful to group rendering that are of the

Reviewer: ```suggestion
suggests, it is meant for dynamic content and is most useful to group objects that are of the
```

## PreRenderedLayer

> Pre-rendered layers are layers that are rendered only once, cached in memory and then just

Reviewer: ```suggestion
Pre-rendered layers are rendered only once, cached in memory and then just
```
